pkg/pod/status.go

```diff
@@ -147,15 +152,26 @@ func MakeTaskRunStatus(logger *zap.SugaredLogger, tr v1beta1.TaskRun, pod *corev
 	return *trs, merr.ErrorOrNil()
 }
 
-func setTaskRunStatusBasedOnStepStatus(logger *zap.SugaredLogger, stepStatuses []corev1.ContainerStatus, tr *v1beta1.TaskRun) *multierror.Error {
+func setTaskRunStatusBasedOnStepStatus(ctx context.Context, logger *zap.SugaredLogger, stepStatuses []corev1.ContainerStatus, tr *v1beta1.TaskRun, kubeclient kubernetes.Interface) *multierror.Error {
 	trs := &tr.Status
 	var merr *multierror.Error
 
+	sidecarLogResults := []v1beta1.PipelineResourceResult{}
+	sidecarLogsResultsEnabled := config.FromContextOrDefaults(ctx).FeatureFlags.ResultExtractionMethod == config.ResultExtractionMethodSidecarLogs
+	if sidecarLogsResultsEnabled && tr.Status.TaskSpec.Results != nil {
+		resultSidecarName := fmt.Sprintf("sidecar-%s", sidecarlogsvalidation.ReservedResultsSidecarName)
+		slr, err := sidecarlogresults.GetResultsFromSidecarLogs(ctx, kubeclient, tr.Namespace, tr.Status.PodName, resultSidecarName)
+		if err != nil {
+			merr = multierror.Append(merr, err)
+		}
+		sidecarLogResults = slr
+	}
 	for _, s := range stepStatuses {
 		if s.State.Terminated != nil && len(s.State.Terminated.Message) != 0 {
 			msg := s.State.Terminated.Message
 
 			results, err := termination.ParseMessage(logger, msg)
+			results = append(results, sidecarLogResults...)
```

Comment:
I'd have thought that it's either results from termination message or results from sidecar logs, not that we need to combine the results from both extraction methods 🤔

Reply:
Its because termination messages can contain other information too. Like Start time etc. We haven't changed that logic here. Its only the results produced by steps that are not in the termination message.

Reply:
ah I see, it may make more sense for this line to go within the else block (after the error check), maybe above line 173? 🤔

Reply:
Agreed. Discussed on GVC. Move line 173 to 188.